tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,586 @@
+import { BooleanLike, classes } from 'common/react';
+import { createSearch } from 'common/string';
+import { flow } from 'common/fp';
+import { map, filter, sortBy } from 'common/collections';
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Tabs, Stack, Box, Input, NoticeBox, Icon } from '../components';
+import { Window } from '../layouts';
+import { Food } from './PreferencesMenu/data';
+
+const TYPE_ICONS = {
+  'Can Make': 'utensils',
+  [Food.Alcohol]: 'wine-glass',
+  [Food.Breakfast]: 'sun',
+  [Food.Bugs]: 'bug',
+  [Food.Cloth]: 'tshirt',
+  [Food.Dairy]: 'cheese',
+  [Food.Fried]: 'fire',
+  [Food.Fruit]: 'apple-alt',
+  [Food.Gore]: 'skull',
+  [Food.Grain]: 'wheat-awn',
+  [Food.Gross]: 'trash',
+  [Food.Junkfood]: 'pizza-slice',
+  [Food.Meat]: 'bacon',
+  [Food.Nuts]: 'seedling',
+  [Food.Oranges]: 'apple-alt',
+  [Food.Pineapple]: 'apple-alt',
+  [Food.Raw]: 'drumstick-bite',
+  [Food.Seafood]: 'fish',
+  [Food.Sugar]: 'candy-cane',
+  [Food.Toxic]: 'biohazard',
+  [Food.Vegetables]: 'carrot',
+};
+
+const CATEGORY_ICONS = {
+  'Can Make': 'utensils',
+  'Breads': 'bread-slice',
+  'Burgers': 'burger',
+  'Cakes': 'cake-candles',
+  'Egg-Based Food': 'egg',
+  'Frozen': 'ice-cream',
+  'Lizard Food': 'dragon',
+  'Meats': 'bacon',
+  'Mexican Food': 'pepper-hot',
+  'Misc. Food': 'bowl-food',
+  'Mothic Food': 'shirt',
+  'Pastries': 'cookie',
+  'Pies': 'chart-pie',
+  'Pizzas': 'pizza-slice',
+  'Salads': 'leaf',
+  'Sandwiches': 'hotdog',
+  'Seafood': 'fish',
+  'Soups': 'mug-saucer',
+  'Spaghettis': 'wheat-awn',
+};
```

Comment:
```suggestion
} as const;
```
same. this locks it as readonly, generally preferred to the enum type

Reply:
Isn't `const TYPE_ICONS = {...` already locking it from editing?

Reply:
![Screenshot 2022-11-28 093456](https://user-images.githubusercontent.com/42397676/204343675-23b8f158-185f-4508-9169-95f62bbbaef5.png)
Nope! Really makes you wonder, sometimes,

Reply:
For the record, we can still reassign things even if they're marked const as const, but it at least gives a little more of a warning, 
![Screenshot 2022-11-28 093939](https://user-images.githubusercontent.com/42397676/204344552-238e02c6-6b98-41e6-8652-8cb9c3be3358.png)

Reply:
In new TS versions this isn't necessary, but we're stuck on 4.3.5 for the moment.